### DISCUSSION TITLE
Fully offline support

### DISCUSSION THREAD

**Discussion by acdha** [github/gh-actions-importer#110] [Feedback] [answered]
What I want is a tool which takes `.travis-ci.yml`, converts it into GitHub Action config, and saves that file in my working directory so I can commit it like normal. 

The current process has a lot of heavyweight parts which in addition to being a lot more work to set up make it fragile and hard to use. For migrating a single project, I had to:

1. Re-enable the Travis CI GitHub app long enough to generate an API token
2. Create a `docker` symlink so it would work with my Podman setup
3. Give up when the `actions-importer/cli` container hit our corporate HTTPS inspection proxy and failed due to not having the internal CA in its trust store.

I'm sure this could be worked through but it'd be nice to have the option of just converting the YAML file and letting me deal with opening a pull-request.

**Comment by j-dunham** [comment]
Hi @acdha have you tried the [--source-file-path](https://docs.github.com/en/actions/migrating-to-github-actions/automated-migrations/migrating-from-travis-ci-with-github-actions-importer#--source-file-path) option?   This option will allow you to transform a local `.travis-ci.yml` file and does not need to make any network requests to TravisCI.  

> Note:  --travis-ci-repository can be whatever you want, it is used to name the workflow in this case.

Here is a example command:
```shell
gh actions-importer dry-run travis-ci --output-dir tmp/dry-run --travis-ci-repository test --source-file-path .travis-ci.yml
```

**Reply by acdha** [reply]
Thanks, that avoided that one problem (I had searched for offline and local). 

`--no-ssl-verify` avoids the proxy issue, although it's somewhat risky compared to using the system CA path, $CURL_CA_BUNDLE, etc.

That leaves the Github access token problem. It looks like it either doesn't or can't reuse the `gh` token and the environmental variable mentioned in the error message doesn't work so I had to specify it on the command-line like this:

```
--github-access-token $GITHUB_ACCESS_TOKEN 
```

**Reply by j-dunham** [reply, marked as answer]
@acdha  are you sure `$GITHUB_ACCESS_TOKEN` is set in the current shell?  If it is then it might be a issue with using Podman because that is untested and it works correctly for me.   

As a work around you can just pass anything for the value, like `--github-access-token testing`. The request to GitHub will fail silently and still transform the pipeline.   If you do this you might be able to remove `--no-ssl-verify` as well but I am not 100% on that though. 

Another option would be to use the [configure](https://docs.github.com/en/actions/migrating-to-github-actions/automated-migrations/automating-migration-with-github-actions-importer#authenticating-at-the-command-line) command to setup a `.env.local` file with the correct value.  This would only be useful if you wanted to use the migrate command to automatically create a PR from you.

**Reply by acdha** [reply]
It was set in the current shell. I suspect that the docker command invocation needed a `-e $GITHUB_ACCESS_TOKEN` but it was easy enough to work around. What I really wanted, however, was for it just to create the file locally since I needed to clean it up later anyway so creating the merge request had minimal value.

**Reply by j-dunham** [reply]
Hi @acdha I just wanted to confirm you are using the [dry-run](https://docs.github.com/en/actions/migrating-to-github-actions/automated-migrations/migrating-from-travis-ci-with-github-actions-importer#perform-a-dry-run-migration-of-a-travis-ci-pipeline) command which should just create the files locally?

**Reply by acdha** [reply]
@j-dunham That worked with a bit of cleanup and I was able to migrate our last projects. Thank you!
